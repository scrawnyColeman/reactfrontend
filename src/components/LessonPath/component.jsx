import React from 'react';
import styled from 'styled-components';
import Button from '../Button/component';
import { colours } from '../../constants/styles';
import { useHistory } from 'react-router-dom';
import { setUserLesson } from '../../data/apiCalls';

const StyledWrapper = styled.div`
    display: flex;
    justify-content: space-between;
    margin auto 0;
`;
const StyledHR = styled.hr`
    border: 1px solid ${colours.primary};
`;
const Container = styled.div`
    max-height: 300px;
    margin: auto;
    width: 75%;
`;
const StyledText = styled.div`
    color: ${colours.secondary};
    font-size: 1.5rem;
    margin: auto 0;
`;

const LessonPath = ({ text, id, path, approved }) => {
    const history = useHistory();
    const user = sessionStorage.getItem('activeId') || null;
    return (
        <Container>
            <StyledWrapper>
                <StyledText>{text}</StyledText>
                <Button
                    text="Begin"
                    onClick={() => {
                        approved && setUserLesson(user, id).catch((error) => console.log(error));
                        history.push({ pathname: `${path}/${id}` });
                    }}
                />
            </StyledWrapper>
            <StyledHR />
        </Container>
    );
};

export default LessonPath;
