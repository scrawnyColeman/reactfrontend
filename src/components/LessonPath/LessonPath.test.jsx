import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import LessonPath from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('LessonPath component', () => {
    const text = 'Java';
    const id = '12345';
    const path = '/path';

    test('component matches snapshot', () => {
        const wrapper = shallow(<LessonPath text={text} id={id} path={path} />);
        expect(wrapper).toMatchSnapshot();
    });
});
