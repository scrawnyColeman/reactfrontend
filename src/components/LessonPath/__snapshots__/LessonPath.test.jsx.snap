// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessonPath component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Java
    </styled.div>
    <Button
      onClick={[Function]}
      text="Begin"
    />
  </styled.div>
  <styled.hr />
</styled.div>
`;
